WordPress/src/main/java/org/wordpress/android/util/SnackbarSequencerInfo.kt

```diff
@@ -0,0 +1,52 @@
+package org.wordpress.android.util
+
+import android.view.View
+import android.view.View.OnClickListener
+import com.google.android.material.snackbar.Snackbar
+import org.wordpress.android.ui.utils.UiString
+import java.lang.ref.SoftReference
+import java.lang.ref.WeakReference
+
+// Taken from com.google.android.material.snackbar.SnackbarManager.java
+// Did not find a way to get them directly from the android framework for now
+private const val SHORT_DURATION_MS = 1500L
+private const val LONG_DURATION_MS = 2750L
+
+const val INDEFINITE_SNACKBAR_NOT_ALLOWED = "Snackbar.LENGTH_INDEFINITE not allowed in getSnackbarDurationMs."
+
+fun getSnackbarDurationMs(snackbarSequencerInfo: SnackbarSequencerInfo): Long {
+    return when (snackbarSequencerInfo.snackbarInfo.duration) {
+        Snackbar.LENGTH_INDEFINITE ->
+            throw IllegalArgumentException(INDEFINITE_SNACKBAR_NOT_ALLOWED)
+        Snackbar.LENGTH_LONG -> LONG_DURATION_MS
+        Snackbar.LENGTH_SHORT -> SHORT_DURATION_MS
+        else -> snackbarSequencerInfo.snackbarInfo.duration.toLong()
+    }
+}
+
+class SnackbarActionInfo(
+    val textRes: UiString,
+    clickListener: OnClickListener
+) {
+    val clickListener = SoftReference(clickListener)
+}
+
+class SnackbarCallbackInfo(
```

Comment:
I think this object is not necessary. Inlining it makes everything more readable. What do you think about something like this: 
```
class SnackbarSequencerInfo(
    val snackbarInfo: SnackbarInfo,
    val snackbarActionInfo: SnackbarActionInfo? = null,
    val dismissCallback: ((transientBottomBar: Snackbar?, event: Int) -> Unit)? = null,
    val shownCallback: ((snackbar: Snackbar?) -> Unit)? = null
) {
    val snackbarCallback: SoftReference<Snackbar.Callback> = SoftReference(object : Snackbar.Callback() {
        override fun onDismissed(transientBottomBar: Snackbar?, event: Int) {
            dismissCallback?.invoke(transientBottomBar, event)
            super.onDismissed(transientBottomBar, event)
        }

        override fun onShown(sb: Snackbar?) {
            super.onShown(sb)
            shownCallback?.invoke(sb)
        }
    })
}
```

Reply:
I chatted a bit also with @malinajirka about this one, let me know if I targeted your intention 😊. I removed the `val` in `val dismissCallback` to try to avoid keeping strong reference (let me know if it makes sense). Changes here 3f60566.

Reply:
looks great! thanks @develric (and good idea with removing of the `val`, I haven't thought of that!)